cobalt/media/media_module.cc

```diff
@@ -135,8 +135,16 @@ class CanPlayTypeHandlerStarboard : public CanPlayTypeHandler {
   SbMediaSupportType CanPlayAdaptive(
       const std::string& mime_type,
       const std::string& key_system) const override {
+    SbMediaSupportType support_type;
     media::FormatSupportQueryMetrics metrics;
-    SbMediaSupportType support_type = CanPlayType(mime_type, key_system);
+    // Only mp4 and webm videos are supported for adaptive playback in Cobalt;
+    // any other containers can be immediately rejected.
+    if (strstr(mime_type.c_str(), "video/mp4") == NULL &&
+        strstr(mime_type.c_str(), "video/webm") == NULL) {
```

Comment:
I think containers other than mp4 and web would be rejected in SbMediaCanPlayMimeAndKeySystem(). If we add an extra check here in cobalt, anytime in the future when we support a new container, we need to modify the code here. What's your consideration to filter containers here?

Reply:
My reasoning here was:
1. To make the logic for adaptive playback match the logic for progressive playback (lines 124-125 above)
2. In theory, starboard shouldn't need to know what container the media is in, because it never handles demuxing. Cobalt itself (or cast, in the case of cast+starboard) is what actually does the demuxing. For example, as-is the default x86 linux implementation of starboard can decode mp3 (since it uses FFmpeg), but cobalt can't (by default) actually demux an mp3 container.

For point 2: I had mentioned this idea to Xiaoming in https://github.com/youtube/cobalt/pull/1135. His response was: "The check of container can (and should) happen in Cobalt. It's now handled in Starboard for legacy reasons. It makes sense to move it from Starboard to Cobalt, but we shouldn't just remove it in Starboard without adding proper support in Cobalt." (see the comment thread there for more context). I'm not sure if anything has changed since then, but that was my reasoning behind this CL.

So basically, this CL is adding the container check to Cobalt. Eventually, the container checks in starboard can likely be removed.

Reply:
I see, thanks for the clarification.

Reply:
I noticed that we made a mistake here. Besides "video/mp4" and "video/webm", we also have "audio/mp4" and "audio/webm". I think let's revert the cl first and discuss on if we should filter container in cobalt more.